04-data-structures-part1.Rmd

```diff
@@ -486,8 +490,40 @@ ncol(matrix_example)
 > that this is typical, and encourage them to take their best guess based on semantic meaning,
 > even if they aren't sure.
 
+> ## Solution to Challenge 3 {.challenge}
+> ```{r, eval=TRUE, echo=TRUE}
+> cats[1]
+> ```
+> We can think of a data frame as a list of vectors. The single brace `[1]` returns the first slice of the list, as another list. In this case it is the first column of the data frame.
+> ```{r, eval=TRUE, echo=TRUE}
+> cats[[1]]
+> ```
+> The double brace `[[1]]` returns the contents of the list item. In this case it is the contents of the first column, a _vector_ of type _factor_.
+> ```{r, eval=TRUE, echo=TRUE}
+> cats$coat
+> ```
+> This example uses the `$` character to address items by name. _coat_ is the first column of the data frame, again a _vector_ of type _factor_.
+> ```{r, eval=TRUE, echo=TRUE}
+> cats["coat"]
+> ```
+> Here we are using a single brace `["coat"]` replacing the index number with the column name. Like example 1, the returned object is a _list_.
+> ```{r, eval=TRUE, echo=TRUE}
+> cats[1,1]
```

Comment:
I think the Google R style guide (which we generally try to follow) recommends a space after a comma. E.g. `cats[1,1]` to `cats[1, 1]` and also in the few lines below (515 and 519) as well.